packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  let ui5Aggregation: UI5Aggregation | undefined = undefined;
+  if (ui5Class === undefined) {
+    ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  }
+
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  // Special case: "class" property can be used on all Controls and selected Elements.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "class") {
+    return [];
+  }
+
+  // Special case: "binding" property is a synonym for "objectBindings" special setting on ManagedObject
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "binding") {
+    return [];
+  }
+
+  // Special case: "stashed" property can appear under sap.uxap.ObjectPageLazyLoader
+  // Technically it can be under any Element but it should only be used for this one.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (
+    attribute.key === "stashed" &&
+    ui5NodeToFQN(ui5Class) === "sap.uxap.ObjectPageLazyLoader"
+  ) {
+    return [];
+  }
+
+  const allProps: (
+    | UI5Prop
+    | UI5Event
+    | UI5Association
+    | UI5Aggregation
+  )[] = flattenProperties(ui5Class);
+  const allEvents = flattenEvents(ui5Class);
+  const allAssociations = flattenAssociations(ui5Class);
+  // Aggregations can be used as attributes for binding
+  const allAggregations = flattenAggregations(ui5Class);
+  const allClassUI5Attributes = allProps
+    .concat(allEvents)
+    .concat(allAssociations)
+    .concat(allAggregations);
+  // TODO Should we filter out non-public/protected names?
+  if (
+    find(allClassUI5Attributes, (_) => _.name === attribute.key) !== undefined
+  ) {
+    return [];
+  }
+
+  return [
+    {
+      kind: "UnknownAttributeKey",
+      message: `Unknown attribute key: ${attribute.key}`,
+      offsetRange: {
+        start: attribute.syntax.key.startOffset,
+        end: attribute.syntax.key.endOffset,
+      },
+      severity: "error",
+    },
+  ];
+}
+
+function isAttributeNameAlwaysValid(attribute: XMLAttribute): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
```

Comment:
may be able to model this at the signature level:

`XMLAttribute & {key:string}`

Reply:
I tried exactly this but typescript didn't let me send the attribute even though I checked the `key` isn't null, and I didn't want to cast it

Reply:
Yeah I just checked this myself, even with newest TypeScript version (3.9.x).
TSC is smart, but not smart enough to understand the previous IF condition as a typeguard
that removes ones of the possible union types.

I think we should handle this with a **cast**, Feels like we are turning a **design time** problem (type inference) into a **runtime problem** (untested branch) so in fact **increasing** the scope of the problem, WDYT?

Reply:
ok

Reply:
Done